packages/engine-multi/test/api/autoinstall.test.ts

```diff
@@ -271,4 +271,114 @@ test.serial('autoinstall: emit an event on completion', async (t) => {
   t.assert(event.duration >= 10);
 });
 
-test.todo('autoinstall: emit on error');
+test.serial('autoinstall: throw on error', async (t) => {
+  const mockIsInstalled = async () => false;
+  const mockInstall = async () => {
+    throw new Error('err');
+  };
+
+  const autoinstallOpts = {
+    handleInstall: mockInstall,
+    handleIsInstalled: mockIsInstalled,
+  };
+  const context = createContext(autoinstallOpts);
+
+  await t.throwsAsync(() => autoinstall(context), {
+    name: 'AutoinstallError',
+    message: 'Error installing @openfn/language-common@1.0.0: err',
+  });
+});
+
+// TODO this is a bit flaky apparently
+test.serial('autoinstall: throw on error twice if pending', async (t) => {
```

Comment:
Yes, this is confirmed flaky.

Maybe this is hinting at whatever underlying problem we have here